### ISSUE TITLE
[Enhancement]: Set compatibility for `container-registry.oracle.com/mysql/community-server` for MySQL module

### ISSUE THREAD

**Issue by linghengqian** [testcontainers/testcontainers-java#9452]
### Module

MySQL

### Proposal

- `mysql` is not the official Docker Image of MySQL. According to https://dev.mysql.com/doc/refman/9.1/en/docker-mysql-getting-started.html and https://github.com/oracle/graalvm-reachability-metadata/pull/350 , only `container-registry.oracle.com/mysql/community-server` is the official image of MySQL. 
- I think it is necessary to set compatibility of `container-registry.oracle.com/mysql/community-server` for MySQL module.

**Comment by haiderkagalwala** [comment]
I'd like to work on this. My approach: add container-registry.oracle.com/mysql/community-server as a second compatible image name in assertCompatibleWith() in MySQLContainer. Does this approach look right to maintainers?

**Comment by linghengqian** [comment]
Perhaps the maintainer doesn't have the time to look into the issues?

**Comment by haiderkagalwala** [comment]
Raised a PR for this: https://github.com/testcontainers/testcontainers-java/pull/11767
The fix adds container-registry.oracle.com/mysql/community-server as a second accepted image name in assertCompatibleWith in both the current org.testcontainers.mysql.MySQLContainer and the deprecated org.testcontainers.containers.MySQLContainer. Happy to get any feedback.

**Comment by eddumelendez** [comment]
I don't think we can add `container-registry.oracle.com/mysql/community-server` as compatible image  if test are failing (`testWithOnlyUserReadableCustomIniFile`, `testMySQLWithCustomIniFile`). This due to a different path for configuration. Also, the purpose of `DockerImageName.parse("container-registry.oracle.com/mysql/community-server:8").asCompatibleSubstituteFor("mysql")` is adding images that developers think are compatible.
